main.js

```diff
@@ -0,0 +1,141 @@
+const alphabet = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"]
+const gallowImages = ["Hangman Images/1.jpg", "Hangman Images/2.jpg", "Hangman Images/3.jpg", "Hangman Images/4.jpg", "Hangman Images/5.jpg", "Hangman Images/6.jpg", "Hangman Images/7.jpg", "Hangman Images/8.jpg", "https://i.ebayimg.com/images/g/jQkAAOSwo-pjeWSL/s-l1600.jpg", "https://i.pinimg.com/236x/db/62/e0/db62e024c5cd86b8a14ec068b2318c85--captain-jack-sparrow-pirates-of-the-caribbean.jpg"];
+let encryptedWord = [];
+let player1 = document.getElementById("word-input");
+let player2 = document.getElementById("word-guessing");
+let wordToGuess;
+let mismatchedLetters = 0;
+let completeBoard = encrypted;
+let alphabetButtons = letters;
+let gallowsImages = images;
+let resetButton = restart;
+
+// hides the page of the player one and displays the elements needed to start the game
+function startGame(player1) {
+    wordToGuess = player1.input.value;
+    switchDisplay();
+    encryptLetters(wordToGuess);
+    statusWord();
+    displayAlphabet();
+    displayImages(mismatchedLetters);
+}
+
+// this is the main function of the game
+// for each clicked button:
+// → the encrypted word will be completed if it is the case
+// → hides the clicked button
+// → checks the status of the game
+// → displays the proper image
+function searchMatches(id) {
+    completeBoard.innerHTML = "";
+    let matchingLetters = completeWord(id);
+    markButton(id);
+    statusWord();
+    countLifes(matchingLetters);
+    displayImages(mismatchedLetters);
+}
+
+// for each letter in the word, a "_" will be displayed
+function encryptLetters(wordToGuess) {
+    for (let i = 0; i < wordToGuess.length; ++i) {
+        encryptedWord.push(" _ ");
+    }
+}
+
+// if all the letters in the word are matched, the game will be over
+// if all the lifes are wasted, the game is also over
+function countLifes(letterMatch) {
+    if (letterMatch == 0) {
+        ++mismatchedLetters;
+        if (mismatchedLetters == 8) {
+            mismatchedLetters = 9;
+            hangmanStatus(mismatchedLetters);
+        }
+    }
+    allLettersMatch(encryptedWord);
+}
+
+// displayes the letters of the alphabet as buttons
+// adds functionality to the buttons
+function displayAlphabet() {
+    for (let i = 0; i < alphabet.length; ++i) {
+        const currentButton = document.createElement("button");
+        alphabetButtons.appendChild(currentButton);
+        currentButton.innerHTML = alphabet[i];
+        currentButton.id = alphabet[i];
+        currentButton.addEventListener("click", ()=> searchMatches(currentButton.id));
+        currentButton.className = "btn btn-light";
+    }
+}
+
+// for each letter in the word, one "-" will be displayed
+function statusWord() {
```

Comment:
What do you need this function for? Why do you need to display "-"?

Reply:
The comment is not suggestive enough. The function rewrites the word at each button press if a letter is found. I changed the comment.